SX1272 LoRa driver is broken and fails to build

**Describe the bug**
We are adding our SX1272-based development board and trying to compile the lora sample application under samples/subsys/lorawan/class_a, it fails to build with the following output:
```
In file included from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:11:
C:/zeph/zephyr/include/sys/atomic_builtin.h:37:20: error: redefinition of 'atomic_cas'
   37 | static inline bool atomic_cas(atomic_t *target, atomic_val_t old_value,
      |                    ^~~~~~~~~~
In file included from C:/zeph/zephyr/include/sys/atomic_c.h:75,
                 from C:/zeph/zephyr/include/sys/atomic.h:30,
                 from C:/zeph/zephyr/include/kernel_includes.h:21,
                 from C:/zeph/zephyr/include/kernel.h:17,
                 from C:/zeph/zephyr/include/init.h:11,
                 from C:/zeph/zephyr/include/device.h:29,
                 from C:/zeph/zephyr/include/drivers/gpio.h:23,
                 from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:8:
C:/zeph/zephyr/samples/subsys/lorawan/class_a/build2/zephyr/include/generated/syscalls/atomic_c.h:28:20: note: previous definition of 'atomic_cas' was here
   28 | static inline bool atomic_cas(atomic_t * target, atomic_val_t old_value, atomic_val_t new_value)
      |                    ^~~~~~~~~~
In file included from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:11:
C:/zeph/zephyr/include/sys/atomic_builtin.h:58:20: error: redefinition of 'atomic_ptr_cas'
   58 | static inline bool atomic_ptr_cas(atomic_ptr_t *target, atomic_ptr_val_t old_value,
      |                    ^~~~~~~~~~~~~~
In file included from C:/zeph/zephyr/include/sys/atomic_c.h:75,
                 from C:/zeph/zephyr/include/sys/atomic.h:30,
                 from C:/zeph/zephyr/include/kernel_includes.h:21,
                 from C:/zeph/zephyr/include/kernel.h:17,
                 from C:/zeph/zephyr/include/init.h:11,
                 from C:/zeph/zephyr/include/device.h:29,
                 from C:/zeph/zephyr/include/drivers/gpio.h:23,
                 from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:8:
C:/zeph/zephyr/samples/subsys/lorawan/class_a/build2/zephyr/include/generated/syscalls/atomic_c.h:42:20: note: previous definition of 'atomic_ptr_cas' was here
   42 | static inline bool atomic_ptr_cas(atomic_ptr_t * target, atomic_ptr_val_t old_value, atomic_ptr_val_t new_value)
      |                    ^~~~~~~~~~~~~~
In file included from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:11:
C:/zeph/zephyr/include/sys/atomic_builtin.h:77:28: error: redefinition of 'atomic_add'
   77 | static inline atomic_val_t atomic_add(atomic_t *target, atomic_val_t value)
      |                            ^~~~~~~~~~
In file included from C:/zeph/zephyr/include/sys/atomic_c.h:75,
                 from C:/zeph/zephyr/include/sys/atomic.h:30,
                 from C:/zeph/zephyr/include/kernel_includes.h:21,
                 from C:/zeph/zephyr/include/kernel.h:17,
                 from C:/zeph/zephyr/include/init.h:11,
                 from C:/zeph/zephyr/include/device.h:29,
                 from C:/zeph/zephyr/include/drivers/gpio.h:23,
                 from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:8:
C:/zeph/zephyr/samples/subsys/lorawan/class_a/build2/zephyr/include/generated/syscalls/atomic_c.h:56:28: note: previous definition of 'atomic_add' was here
   56 | static inline atomic_val_t atomic_add(atomic_t * target, atomic_val_t value)
      |                            ^~~~~~~~~~
In file included from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:11:
C:/zeph/zephyr/include/sys/atomic_builtin.h:93:28: error: redefinition of 'atomic_sub'
   93 | static inline atomic_val_t atomic_sub(atomic_t *target, atomic_val_t value)
      |                            ^~~~~~~~~~
In file included from C:/zeph/zephyr/include/sys/atomic_c.h:75,
                 from C:/zeph/zephyr/include/sys/atomic.h:30,
                 from C:/zeph/zephyr/include/kernel_includes.h:21,
                 from C:/zeph/zephyr/include/kernel.h:17,
                 from C:/zeph/zephyr/include/init.h:11,
                 from C:/zeph/zephyr/include/device.h:29,
                 from C:/zeph/zephyr/include/drivers/gpio.h:23,
                 from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:8:
C:/zeph/zephyr/samples/subsys/lorawan/class_a/build2/zephyr/include/generated/syscalls/atomic_c.h:70:28: note: previous definition of 'atomic_sub' was here
   70 | static inline atomic_val_t atomic_sub(atomic_t * target, atomic_val_t value)
      |                            ^~~~~~~~~~
In file included from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:11:
C:/zeph/zephyr/include/sys/atomic_builtin.h:108:28: error: redefinition of 'atomic_inc'
  108 | static inline atomic_val_t atomic_inc(atomic_t *target)
      |                            ^~~~~~~~~~
In file included from C:/zeph/zephyr/include/sys/atomic.h:30,
                 from C:/zeph/zephyr/include/kernel_includes.h:21,
                 from C:/zeph/zephyr/include/kernel.h:17,
                 from C:/zeph/zephyr/include/init.h:11,
                 from C:/zeph/zephyr/include/device.h:29,
                 from C:/zeph/zephyr/include/drivers/gpio.h:23,
                 from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:8:
C:/zeph/zephyr/include/sys/atomic_c.h:30:28: note: previous definition of 'atomic_inc' was here
   30 | static inline atomic_val_t atomic_inc(atomic_t *target)
      |                            ^~~~~~~~~~
In file included from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:11:
C:/zeph/zephyr/include/sys/atomic_builtin.h:123:28: error: redefinition of 'atomic_dec'
  123 | static inline atomic_val_t atomic_dec(atomic_t *target)
      |                            ^~~~~~~~~~
In file included from C:/zeph/zephyr/include/sys/atomic.h:30,
                 from C:/zeph/zephyr/include/kernel_includes.h:21,
                 from C:/zeph/zephyr/include/kernel.h:17,
                 from C:/zeph/zephyr/include/init.h:11,
                 from C:/zeph/zephyr/include/device.h:29,
                 from C:/zeph/zephyr/include/drivers/gpio.h:23,
                 from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:8:
C:/zeph/zephyr/include/sys/atomic_c.h:36:28: note: previous definition of 'atomic_dec' was here
   36 | static inline atomic_val_t atomic_dec(atomic_t *target)
      |                            ^~~~~~~~~~
In file included from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:11:
C:/zeph/zephyr/include/sys/atomic_builtin.h:138:28: error: static declaration of 'atomic_get' follows non-static declaration
  138 | static inline atomic_val_t atomic_get(const atomic_t *target)
      |                            ^~~~~~~~~~
In file included from C:/zeph/zephyr/include/sys/atomic.h:30,
                 from C:/zeph/zephyr/include/kernel_includes.h:21,
                 from C:/zeph/zephyr/include/kernel.h:17,
                 from C:/zeph/zephyr/include/init.h:11,
                 from C:/zeph/zephyr/include/device.h:29,
                 from C:/zeph/zephyr/include/drivers/gpio.h:23,
                 from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:8:
C:/zeph/zephyr/include/sys/atomic_c.h:42:21: note: previous declaration of 'atomic_get' was here
   42 | extern atomic_val_t atomic_get(const atomic_t *target);
      |                     ^~~~~~~~~~
In file included from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:11:
C:/zeph/zephyr/include/sys/atomic_builtin.h:153:32: error: static declaration of 'atomic_ptr_get' follows non-static declaration
  153 | static inline atomic_ptr_val_t atomic_ptr_get(const atomic_ptr_t *target)
      |                                ^~~~~~~~~~~~~~
In file included from C:/zeph/zephyr/include/sys/atomic.h:30,
                 from C:/zeph/zephyr/include/kernel_includes.h:21,
                 from C:/zeph/zephyr/include/kernel.h:17,
                 from C:/zeph/zephyr/include/init.h:11,
                 from C:/zeph/zephyr/include/device.h:29,
                 from C:/zeph/zephyr/include/drivers/gpio.h:23,
                 from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:8:
C:/zeph/zephyr/include/sys/atomic_c.h:44:25: note: previous declaration of 'atomic_ptr_get' was here
   44 | extern atomic_ptr_val_t atomic_ptr_get(const atomic_ptr_t *target);
      |                         ^~~~~~~~~~~~~~
In file included from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:11:
C:/zeph/zephyr/include/sys/atomic_builtin.h:170:28: error: redefinition of 'atomic_set'
  170 | static inline atomic_val_t atomic_set(atomic_t *target, atomic_val_t value)
      |                            ^~~~~~~~~~
In file included from C:/zeph/zephyr/include/sys/atomic_c.h:75,
                 from C:/zeph/zephyr/include/sys/atomic.h:30,
                 from C:/zeph/zephyr/include/kernel_includes.h:21,
                 from C:/zeph/zephyr/include/kernel.h:17,
                 from C:/zeph/zephyr/include/init.h:11,
                 from C:/zeph/zephyr/include/device.h:29,
                 from C:/zeph/zephyr/include/drivers/gpio.h:23,
                 from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:8:
C:/zeph/zephyr/samples/subsys/lorawan/class_a/build2/zephyr/include/generated/syscalls/atomic_c.h:84:28: note: previous definition of 'atomic_set' was here
   84 | static inline atomic_val_t atomic_set(atomic_t * target, atomic_val_t value)
      |                            ^~~~~~~~~~
In file included from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:11:
C:/zeph/zephyr/include/sys/atomic_builtin.h:191:32: error: redefinition of 'atomic_ptr_set'
  191 | static inline atomic_ptr_val_t atomic_ptr_set(atomic_ptr_t *target, atomic_ptr_val_t value)
      |                                ^~~~~~~~~~~~~~
In file included from C:/zeph/zephyr/include/sys/atomic_c.h:75,
                 from C:/zeph/zephyr/include/sys/atomic.h:30,
                 from C:/zeph/zephyr/include/kernel_includes.h:21,
                 from C:/zeph/zephyr/include/kernel.h:17,
                 from C:/zeph/zephyr/include/init.h:11,
                 from C:/zeph/zephyr/include/device.h:29,
                 from C:/zeph/zephyr/include/drivers/gpio.h:23,
                 from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:8:
C:/zeph/zephyr/samples/subsys/lorawan/class_a/build2/zephyr/include/generated/syscalls/atomic_c.h:98:32: note: previous definition of 'atomic_ptr_set' was here
   98 | static inline atomic_ptr_val_t atomic_ptr_set(atomic_ptr_t * target, atomic_ptr_val_t value)
      |                                ^~~~~~~~~~~~~~
In file included from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:11:
C:/zeph/zephyr/include/sys/atomic_builtin.h:207:28: error: redefinition of 'atomic_clear'
  207 | static inline atomic_val_t atomic_clear(atomic_t *target)
      |                            ^~~~~~~~~~~~
In file included from C:/zeph/zephyr/include/sys/atomic.h:30,
                 from C:/zeph/zephyr/include/kernel_includes.h:21,
                 from C:/zeph/zephyr/include/kernel.h:17,
                 from C:/zeph/zephyr/include/init.h:11,
                 from C:/zeph/zephyr/include/device.h:29,
                 from C:/zeph/zephyr/include/drivers/gpio.h:23,
                 from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:8:
C:/zeph/zephyr/include/sys/atomic_c.h:50:28: note: previous definition of 'atomic_clear' was here
   50 | static inline atomic_val_t atomic_clear(atomic_t *target)
      |                            ^~~~~~~~~~~~
In file included from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:11:
C:/zeph/zephyr/include/sys/atomic_builtin.h:223:32: error: redefinition of 'atomic_ptr_clear'
  223 | static inline atomic_ptr_val_t atomic_ptr_clear(atomic_ptr_t *target)
      |                                ^~~~~~~~~~~~~~~~
In file included from C:/zeph/zephyr/include/sys/atomic.h:30,
                 from C:/zeph/zephyr/include/kernel_includes.h:21,
                 from C:/zeph/zephyr/include/kernel.h:17,
                 from C:/zeph/zephyr/include/init.h:11,
                 from C:/zeph/zephyr/include/device.h:29,
                 from C:/zeph/zephyr/include/drivers/gpio.h:23,
                 from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:8:
C:/zeph/zephyr/include/sys/atomic_c.h:56:32: note: previous definition of 'atomic_ptr_clear' was here
   56 | static inline atomic_ptr_val_t atomic_ptr_clear(atomic_ptr_t *target)
      |                                ^~~~~~~~~~~~~~~~
In file included from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:11:
C:/zeph/zephyr/include/sys/atomic_builtin.h:240:28: error: redefinition of 'atomic_or'
  240 | static inline atomic_val_t atomic_or(atomic_t *target, atomic_val_t value)
      |                            ^~~~~~~~~
In file included from C:/zeph/zephyr/include/sys/atomic_c.h:75,
                 from C:/zeph/zephyr/include/sys/atomic.h:30,
                 from C:/zeph/zephyr/include/kernel_includes.h:21,
                 from C:/zeph/zephyr/include/kernel.h:17,
                 from C:/zeph/zephyr/include/init.h:11,
                 from C:/zeph/zephyr/include/device.h:29,
                 from C:/zeph/zephyr/include/drivers/gpio.h:23,
                 from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:8:
C:/zeph/zephyr/samples/subsys/lorawan/class_a/build2/zephyr/include/generated/syscalls/atomic_c.h:112:28: note: previous definition of 'atomic_or' was here
  112 | static inline atomic_val_t atomic_or(atomic_t * target, atomic_val_t value)
      |                            ^~~~~~~~~
In file included from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:11:
C:/zeph/zephyr/include/sys/atomic_builtin.h:257:28: error: redefinition of 'atomic_xor'
  257 | static inline atomic_val_t atomic_xor(atomic_t *target, atomic_val_t value)
      |                            ^~~~~~~~~~
In file included from C:/zeph/zephyr/include/sys/atomic_c.h:75,
                 from C:/zeph/zephyr/include/sys/atomic.h:30,
                 from C:/zeph/zephyr/include/kernel_includes.h:21,
                 from C:/zeph/zephyr/include/kernel.h:17,
                 from C:/zeph/zephyr/include/init.h:11,
                 from C:/zeph/zephyr/include/device.h:29,
                 from C:/zeph/zephyr/include/drivers/gpio.h:23,
                 from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:8:
C:/zeph/zephyr/samples/subsys/lorawan/class_a/build2/zephyr/include/generated/syscalls/atomic_c.h:126:28: note: previous definition of 'atomic_xor' was here
  126 | static inline atomic_val_t atomic_xor(atomic_t * target, atomic_val_t value)
      |                            ^~~~~~~~~~
In file included from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:11:
C:/zeph/zephyr/include/sys/atomic_builtin.h:274:28: error: redefinition of 'atomic_and'
  274 | static inline atomic_val_t atomic_and(atomic_t *target, atomic_val_t value)
      |                            ^~~~~~~~~~
In file included from C:/zeph/zephyr/include/sys/atomic_c.h:75,
                 from C:/zeph/zephyr/include/sys/atomic.h:30,
                 from C:/zeph/zephyr/include/kernel_includes.h:21,
                 from C:/zeph/zephyr/include/kernel.h:17,
                 from C:/zeph/zephyr/include/init.h:11,
                 from C:/zeph/zephyr/include/device.h:29,
                 from C:/zeph/zephyr/include/drivers/gpio.h:23,
                 from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:8:
C:/zeph/zephyr/samples/subsys/lorawan/class_a/build2/zephyr/include/generated/syscalls/atomic_c.h:140:28: note: previous definition of 'atomic_and' was here
  140 | static inline atomic_val_t atomic_and(atomic_t * target, atomic_val_t value)
      |                            ^~~~~~~~~~
In file included from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:11:
C:/zeph/zephyr/include/sys/atomic_builtin.h:291:28: error: redefinition of 'atomic_nand'
  291 | static inline atomic_val_t atomic_nand(atomic_t *target, atomic_val_t value)
      |                            ^~~~~~~~~~~
In file included from C:/zeph/zephyr/include/sys/atomic_c.h:75,
                 from C:/zeph/zephyr/include/sys/atomic.h:30,
                 from C:/zeph/zephyr/include/kernel_includes.h:21,
                 from C:/zeph/zephyr/include/kernel.h:17,
                 from C:/zeph/zephyr/include/init.h:11,
                 from C:/zeph/zephyr/include/device.h:29,
                 from C:/zeph/zephyr/include/drivers/gpio.h:23,
                 from C:/zeph/zephyr/drivers/lora/sx12xx_common.c:8:
C:/zeph/zephyr/samples/subsys/lorawan/class_a/build2/zephyr/include/generated/syscalls/atomic_c.h:154:28: note: previous definition of 'atomic_nand' was here
  154 | static inline atomic_val_t atomic_nand(atomic_t * target, atomic_val_t value)
      |                            ^~~~~~~~~~~
[49/161] Building C object zephyr/subsys/lorawan/CMakeFile...dir/C_/zeph/modules/lib/loramac-node/src/mac/LoRaMac.c.obj
ninja: build stopped: subcommand failed.
```

**To Reproduce**
Steps to reproduce the behavior: Build an SX1272-based board with the lorawan sample app

**Expected behavior**
For the compilation to complete

**Impact**
Showstopper

**Environment (please complete the following information):**
 - OS: Windows 10
 - Toolchain: GNU Arm Embedded Toolchain 10.3-2021.07
 - Commit SHA or Version used: b6a3dc8d715d0cdb6edeb9e144f2687fd569cb16

--- a/drivers/lora/sx12xx_common.c
+++ b/drivers/lora/sx12xx_common.c
@@ -8,7 +8,7 @@
 #include <drivers/gpio.h>
 #include <drivers/lora.h>
 #include <logging/log.h>
-#include <sys/atomic_builtin.h>
+#include <sys/atomic.h>
 #include <zephyr.h>
 
 /* LoRaMac-node specific includes */

